feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/utils/tabs/models/PageSession.kt

```diff
@@ -0,0 +1,88 @@
+package io.novafoundation.nova.feature_dapp_impl.utils.tabs.models
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.webkit.WebView
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.PageCallback
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3ChromeClient
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3WebViewClient
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.injectWeb3
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.uninjectWeb3
+import java.util.Date
+
+class PageSessionFactory {
+
+    fun create(tabId: String, sessionStartTime: Date, startUrl: String): PageSession {
+        return PageSession(
+            tabId = tabId,
+            sessionStartTime = sessionStartTime,
+            startUrl = startUrl
+        )
+    }
+}
+
+class PageSession(
+    val tabId: String,
+    val sessionStartTime: Date,
+    val startUrl: String
+) : PageCallback {
+
+    private var _webView: WebView? = null
+    val webView: WebView
+        get() {
+            return _webView ?: throw IllegalStateException("WebView is not initialized")
+        }
+
+    private var _webViewClient: Web3WebViewClient? = null
+    val webViewClient: Web3WebViewClient
+        get() {
+            return _webViewClient ?: throw IllegalStateException("WebViewClient is not initialized")
+        }
```

Comment:
The approach of post-initialization (after constructor invocation) is bad since it leaves the object in incosistent state for some time.
If PageSession needs fragment context to properly function, it should be created inside the fragment
If Activity context us enough, we can use `ContextManager` in the factory to obtain activity context